lib/matplotlib/tests/test_cycles.py

```diff
@@ -1,140 +1,100 @@
-import platform
-
-from matplotlib.testing.decorators import image_comparison
+import matplotlib as mpl
 import matplotlib.pyplot as plt
 import numpy as np
 import pytest
 
 from cycler import cycler
 
 
-@image_comparison(['color_cycle_basic.png'], remove_text=True,
-                  tol={'aarch64': 0.02}.get(platform.machine(), 0.0))
 def test_colorcycle_basic():
     fig, ax = plt.subplots()
     ax.set_prop_cycle(cycler('color', ['r', 'g', 'y']))
-    xs = np.arange(10)
-    ys = 0.25 * xs + 2
-    ax.plot(xs, ys, label='red', lw=4)
-    ys = 0.45 * xs + 3
-    ax.plot(xs, ys, label='green', lw=4)
-    ys = 0.65 * xs + 4
-    ax.plot(xs, ys, label='yellow', lw=4)
-    ys = 0.85 * xs + 5
-    ax.plot(xs, ys, label='red2', lw=4)
-    ax.legend(loc='upper left')
-
-
-@image_comparison(['marker_cycle.png', 'marker_cycle.png'], remove_text=True,
-                  tol={'aarch64': 0.02}.get(platform.machine(), 0.0))
+    for _ in range(4):
+        ax.plot(range(10), range(10))
+    assert [l.get_color() for l in ax.lines] == ['r', 'g', 'y', 'r']
+
+
 def test_marker_cycle():
     fig, ax = plt.subplots()
     ax.set_prop_cycle(cycler('c', ['r', 'g', 'y']) +
                       cycler('marker', ['.', '*', 'x']))
-    xs = np.arange(10)
-    ys = 0.25 * xs + 2
-    ax.plot(xs, ys, label='red dot', lw=4, ms=16)
-    ys = 0.45 * xs + 3
-    ax.plot(xs, ys, label='green star', lw=4, ms=16)
-    ys = 0.65 * xs + 4
-    ax.plot(xs, ys, label='yellow x', lw=4, ms=16)
-    ys = 0.85 * xs + 5
-    ax.plot(xs, ys, label='red2 dot', lw=4, ms=16)
-    ax.legend(loc='upper left')
+    for _ in range(4):
+        ax.plot(range(10), range(10))
+    assert [l.get_color() for l in ax.lines] == ['r', 'g', 'y', 'r']
+    assert [l.get_marker() for l in ax.lines] == ['.', '*', 'x', '.']
 
     fig, ax = plt.subplots()
```

Comment:
separate into new test or parameterize this with the typing?

Reply:
It was one test before, and isn't even that easy to parametrize (I think?).

Reply:
It should still be two tests since it's testing different things, very much so since you're right about it being hard to parameterize. I missed that the first test concats two cycler objects:
```python
ax.set_prop_cycle(cycler('c', ['r', 'g', 'y']) + cycler('marker', ['.', '*', 'x']))
```
and the second cleanly separates them:
```python
ax.set_prop_cycle(c=np.array(['r', 'g', 'y']), marker=iter(['.', '*', 'x']))
```

Reply:
ok, split them